### PR TITLE
f/wpc 538 change CTA richtext field to string input field

### PR DESCRIPTION
**Pull request by lakeside763**
Test: https://editor.fwpc538co.wepublish.dev/peering

PR: New Features
Convert callToActionText richtext field to string input field
https://wepublish.atlassian.net/browse/WPC-538

rictext_callToAction
string_call_to_action

### CODE REVIEW

File: examples/website/src/shared/route/articleTemplateContainer.tsx [hunk 1]
```diff
@@ -419,7 +419,7 @@ export function PeerProfileBlock({peer, article}: PeerProfileBlockProps) {
         <div className={css(PeerProfileFiller)} />
       </div>
       <div className={css(PeerProfileCallToActionURL)}>
-        {peer?.callToActionText?.length && peer?.callToActionText?.length > 0 ? (
+        {peer?.callToActionText?.length ? (
```
Comment (roosnic1) [inline examples/website/src/shared/route/articleTemplateContainer.tsx:422]: both a `string` and an `array` have the property `length`. Maybe use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray

File: examples/website/src/shared/types.ts [hunk 2]
```diff
@@ -157,7 +157,7 @@ export interface Peer {
   websiteURL: string
   themeColor: string
   themeFontColor: string
-  callToActionText: any
+  callToActionText: Node[]
```
Comment (roosnic1) [inline examples/website/src/shared/types.ts]: Why `Node[]`? Was it any before you changed it? This should be `string | Node[]`

File: packages/api-db-mongodb/src/migration.ts [hunk 3]
```diff
@@ -619,27 +619,22 @@ export const Migrations: Migration[] = [
...
-          }
-          return peerProfile
-        })
+      const peerProfile = await db.collection(CollectionName.PeerProfiles).findOne({
+        callToActionText: {$exists: true}
+      })
+      if (peerProfile) {
+        await db.collection(CollectionName.PeerProfiles).updateOne(
+          {
+            callToActionText: {$exists: true}
```
Comment (roosnic1) [inline packages/api-db-mongodb/src/migration.ts:628]: here it's better if you use the `_id` from `peerProfile` you got on line 622.

File: packages/api-db-mongodb/src/migration.ts [hunk 4]
```diff
@@ -619,27 +619,22 @@ export const Migrations: Migration[] = [
...
+          },
+          {
+            $set: {
+              callToActionText: richTextToString('', peerProfile?.callToActionText)
+            }
+          },
+          {upsert: true}
```
Comment (roosnic1) [inline packages/api-db-mongodb/src/migration.ts:635]: why the `upsert` option?

File: packages/api-db-mongodb/src/utility.ts [hunk 5]
```diff
@@ -40,7 +40,13 @@ export function richTextToString(blocksString: string, nodes: any = []) {
   const richTextNode = nodes.reduce((string: string, node: any) => {
     if (!node.text && !node.children) return string
     if (node.text) {
-      return `${string} ${node.text}`
+      if (string.endsWith(' ') || node.text.startsWith(' ')) {
```
Comment (roosnic1) [inline packages/api-db-mongodb/src/utility.ts:43]: Isn't this here a usecase for https://www.javascripttutorial.net/javascript-string-trim/

**Review by roosnic1 — CHANGES_REQUESTED** [review]

File: packages/api-db-mongodb/src/utility.ts (hunk 5 above)
Comment (bishoymelek) [reply packages/api-db-mongodb/src/utility.ts:43]: @roosnic1 can you elaborate?

File: packages/editor/src/client/panel/peerProfileEditPanel.tsx [hunk 6]
```diff
@@ -104,7 +98,7 @@ export function PeerInfoEditPanel({onClose, onSave}: ImageEditPanelProps) {
           logoID: logoImage?.id,
           themeColor,
           themeFontColor,
-          callToActionText,
+          callToActionText: callToActionText,
```
Comment (bishoymelek) [inline packages/editor/src/client/panel/peerProfileEditPanel.tsx:101]: can you clean it up?

File: packages/editor/src/client/panel/peerProfileEditPanel.tsx [hunk 7]
```diff
@@ -175,7 +169,12 @@ export function PeerInfoEditPanel({onClose, onSave}: ImageEditPanelProps) {
               }}>
               <FormGroup>
                 <ControlLabel>{t('peerList.panels.text')}</ControlLabel>
-                <RichTextBlock value={callToActionText} onChange={setCallToActionText} />
+                <FormControl
+                  placeholder={t('peerList.panels.text')}
+                  name="callToActionText"
+                  value={callToActionText}
+                  onChange={text => setCallToActionText(text)}
```
Comment (bishoymelek) [inline packages/editor/src/client/panel/peerProfileEditPanel.tsx:176]: maybe something like:
```
onChange={text => setCallToActionText(text)}
```

File: packages/api-db-mongodb/src/migration.ts [hunk 8]
```diff
@@ -613,6 +614,28 @@ export const Migrations: Migration[] = [
...
+        await db.collection(CollectionName.PeerProfiles).updateOne(
+          {
+            _id: peerProfile._id
+          },
+          {
+            $set: {
+              callToActionText: richTextToString('', peerProfile?.callToActionText)
+            }
+          },
+          {upsert: false}
```
Comment (bishoymelek) [inline packages/api-db-mongodb/src/migration.ts:635]: I think default value is false.

**Review by bishoymelek — COMMENTED** [review]
Great one, just very small suggestions.

**Review by bishoymelek — COMMENTED** [review]
I have same review points as before.

**Review by roosnic1 — DISMISSED** [review]
Please check @bishoymelek review

**Review by bishoymelek — APPROVED** [review]
:tada:

**Review by bishoymelek — APPROVED** [review]
:tada:

**Pull request closed without merging**